packages/components/icons/src/templates/icon.styles.tsx

```diff
@@ -56,17 +56,39 @@ const getSizeStyle = (size: Props['size']) => {
   }
 };
 
-const capitalize = (value: string) => value[0].toUpperCase() + value.slice(1);
-
 const getColor = (color: Props['color'], theme: Theme) => {
   if (!color) return 'inherit';
-  const overwrittenVars = {
-    ...vars,
-    ...theme,
-  };
 
-  // @ts-expect-error
-  const iconColor = overwrittenVars[`color${capitalize(color)}`];
```

Comment:
To avoid having to ignore the error, we explicitly map the supported colors to the correct design token.